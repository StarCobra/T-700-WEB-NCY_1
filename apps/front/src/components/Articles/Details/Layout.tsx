import React from "react";
import "../../../style/article.scss";
import ArticleContent from "./ArticleContent";
import { Box, Button } from "@mui/material";
import { useArticle } from "../../../Context/ArticleProvider";
import { Link } from "react-router-dom";

export default function Layout() {
  const { currentArticle } = useArticle();
  console.log("currentArticle", currentArticle);
  return (
    <Box className="fullArticleDisplay">
      <Box className="articleHeader">
        <Box className="articleThumbnail">
          <img src={currentArticle?.enclosure?.[0]?.$?.url} alt="" />
        </Box>

        <Box className="articleTitleContainer">
          <Box className="title">
            <h3>{currentArticle?.title}</h3>
          </Box>

          <Box className="dateAndSource">
            <h4>{currentArticle?.["dc:creator"]}</h4>
          </Box>
        </Box>
      </Box>

      <ArticleContent article={currentArticle} />

      <Link className="backButton" to={"/articles"}>
        <Button>Back</Button>
      </Link>
    </Box>
  );
}
