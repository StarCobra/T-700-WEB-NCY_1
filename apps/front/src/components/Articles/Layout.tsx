import React from "react";
import "../../style/article.scss";
import OneArticle from "./OneArticle";
import { Box } from "@mui/material";

export default function Layout() {
  return (
    <Box className="articleMasterContainer">
      <Box className="articleSampleContainer">
        <OneArticle />
      </Box>

      <Box className="articleSampleContainer">
        <OneArticle />
      </Box>

      <Box className="articleSampleContainer">
        <OneArticle />
      </Box>

      <Box className="articleSampleContainer">
        <OneArticle />
      </Box>
    </Box>
  );
}
