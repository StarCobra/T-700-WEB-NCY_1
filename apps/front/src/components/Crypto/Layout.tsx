import React from "react";
import CryptoArray from "../Crypto/Array";
import Select from "../Select";
import { Box } from "@mui/material";

export default function Layout() {
  const options = [
    { value: "BTC", label: "BTC" },
    { value: "ETH", label: "ETH" },
    { value: "SOL", label: "SOL" },
    { value: "ADA", label: "ADA" },
    { value: "DOT", label: "DOT" },
    { value: "LUNA", label: "LUNA" },
    { value: "LINK", label: "LINK" },
  ];

  return (
    <Box className="ArrayDisplay">
      <Select
        name="Select"
        label="Select a crypto(s):"
        type="multiple"
        options={options}
        // eslint-disable-next-line prettier/prettier
        handleChange={(e: any) => console.log(e.target.value)}
      />

      <CryptoArray />
    </Box>
  );
}
