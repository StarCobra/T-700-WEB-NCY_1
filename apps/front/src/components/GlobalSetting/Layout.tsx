import * as React from "react";
import "../../style/admin.scss";
import { Box } from "@mui/material";
import SettingsIcon from "@mui/icons-material/Settings";
import ManagementCrypto from "./ManagementCrypto";
import useAuth from "../../Context/UserProvider";
import ManagementKeyword from "./ManagementKeyword";

export default function Layout() {
  const { userToken } = useAuth();

  return (
    <Box className="ArrayDisplay">
      <Box className="titleContainer">
        <Box className="title">
          <h1>Global Settings</h1>
        </Box>
        <Box className="icon">
          <SettingsIcon />
        </Box>
      </Box>

      <Box>
        <ManagementCrypto userToken={userToken} />

        <ManagementKeyword userToken={userToken} />
      </Box>

      <Box className="commentContainer">
        <h3>
          {"Please "}
          <a href="mailto:developpers@cryptofanatix.com?subject=Request&body=Hello, I have a request for you, ">
            contact a developper
          </a>
          {" if you have an idea or a problem."}
        </h3>
      </Box>
    </Box>
  );
}
