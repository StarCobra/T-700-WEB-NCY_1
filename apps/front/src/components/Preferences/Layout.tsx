import React from "react";
import { Box } from "@mui/material";
import CryptoArray from "../Crypto/Array";
import Select from "../Select";
import "../../style/preferencies.scss";
import "../../style/cryptoDisplay.scss";
import api from "../../services/api";
import Loader from "../Loader";

export default function Layout() {
  const [favoriteCrypto, setFavoriteCrypto] = React.useState([]);

  const [responseAllCrypto, setResponseAllCrypto] = React.useState([] as any);
  const [loadingAllCrypto, setLoadingAllCrypto] = React.useState(true);

  // Requete pour récupérer toutes les cryptos
  React.useEffect(() => {
    const fetchData = async () => {
      try {
        setLoadingAllCrypto(true);
        const data = await api.getAllCryptos();
        setResponseAllCrypto(data);
      } catch (error) {
        console.error("Error fetching cryptos:", error);
      } finally {
        setLoadingAllCrypto(false);
      }
    };

    fetchData();
  }, []);

  return (
    <Box className="preferencesLayout">
      <Box className="headerContainer">
        <h3>Preferences page</h3>
        <small>
          This page is used to manage your preferences. You can choose your
          favorite crypto.
        </small>
        <small>
          The chosen crypto will be displayed on the home page and you&apos;ll
          receive in priority the new articles about it.
        </small>
      </Box>
      <Box className="arrayContainer">
        {loadingAllCrypto ? (
          <Loader />
        ) : (
          <>
            <h3>Table of cryptos available</h3>
            <CryptoArray resource={responseAllCrypto.data} />
          </>
        )}
      </Box>
      <Box className="addPrefContainer">
        <Box className="selectContainer">
          <h3>Add crypto to your favorite cypto</h3>
          <Select
            name="Select"
            label={
              loadingAllCrypto
                ? "Loading crypto..."
                : "Favorite crypto(s) chosen:"
            }
            type="multiple"
            options={responseAllCrypto.data}
            forCrypto={true}
            handleChange={(e: any) => setFavoriteCrypto(e.target.value)}
          />
        </Box>
        <Box className="cryptoDisplayContainer">
          <h3>
            You have chosen {favoriteCrypto.length} crypto(s) as favorite crypto
            :
          </h3>
          {favoriteCrypto.length > 0 && (
            <ul>
              {favoriteCrypto.map((crypto: any) => (
                <li key={crypto}>{crypto}</li>
              ))}
            </ul>
          )}
        </Box>
      </Box>
    </Box>
  );
}
