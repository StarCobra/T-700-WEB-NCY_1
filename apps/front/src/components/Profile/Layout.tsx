import React from "react";
import "../../style/profile.scss";
import ProfileInformation from "./Information";
import { Box } from "@mui/material";
import exampleImage from "../../assets/profile/profileExample.png";

export default function Layout() {
  return (
    <Box className="centeringDiv">
      <Box className="profileLayout">
        <Box className="profileThumbnailContainer">
          <img src={exampleImage} alt="" />
        </Box>

        <ProfileInformation />
      </Box>
    </Box>
  );
}
