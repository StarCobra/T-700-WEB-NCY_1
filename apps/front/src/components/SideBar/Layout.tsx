import React from "react";
import User from "./User";
import Menu from "./Menu";
import GuestMenu from "./GuestMenu";
import GuestUser from "./GuestUser";
import "../../style/Default/default.scss";
import "../../style/sidebar.scss";
import { Box } from "@mui/material";
import Logo from "../../assets/sideBar/logo.png";
import { Link } from "react-router-dom";

export default function Layout() {
  // const defaultUser = {name:"ArnaudBrg", mail:"ArnaudBrg@gmail.com"}
  const defaultUser = null
  if (defaultUser != null) {
    return (
      <Box className="sideBarContainer">
        <Link to="/" className="logoLocation">
          <img src={Logo} alt="" />
        </Link>

        <User user={defaultUser} />

        <Menu />
      </Box>
    );  
  }

  else {
    return (
      <Box className="sideBarContainer">
        <Link to="/" className="logoLocation">
          <img src={Logo} alt="" />
        </Link>

        <GuestUser />

        <GuestMenu />
      </Box>
    );
  }
  
}
