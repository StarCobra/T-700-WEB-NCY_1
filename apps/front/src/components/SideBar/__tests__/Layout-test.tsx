import React from "react";
import { act, create } from "react-test-renderer";
import { MemoryRouter } from "react-router-dom";
import Layout from "../Layout";

describe("Components > SideBar > Layout", () => {
  const tests = [
    {
      description: "generic",
      props: {},
    },
  ];
  tests.forEach((test) => {
    it(`should render the component ${test.description}`, async () => {
      let root;
      await act(async () => {
        root = create(
          <MemoryRouter>
            <Layout {...test.props} />
          </MemoryRouter>,
        );
      });
      expect(root).toMatchSnapshot();
    });
  });
});
