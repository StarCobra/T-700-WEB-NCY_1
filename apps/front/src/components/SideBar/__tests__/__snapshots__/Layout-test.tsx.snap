// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components > SideBar > Layout should render the component generic 1`] = `
<div
  className="sideBarContainer MuiBox-root css-0"
>
  <div
    className="logoLocation MuiBox-root css-0"
  >
    <img
      alt=""
      src="default"
    />
  </div>
  <div
    className="userLocation MuiBox-root css-0"
  >
    <div
      className="userProfilePicture MuiBox-root css-0"
    >
      <img
        alt=""
        src="default"
      />
    </div>
    <div
      className="usernameContainer MuiBox-root css-0"
    >
      <div
        className="userName MuiBox-root css-0"
      >
        Arnaud B
      </div>
      <div
        className="userMail MuiBox-root css-0"
      >
        arnaud.bourgoin@gmail.com
      </div>
    </div>
  </div>
  <div
    className="differentMenus MuiBox-root css-0"
  >
    <div
      className="Menu MuiBox-root css-0"
    >
      <div
        className="menuIcon MuiBox-root css-0"
      >
        <svg
          className="menuIcon"
          fill="none"
          stroke="currentColor"
          strokeWidth={1.5}
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M2.25 12l8.954-8.955c.44-.439 1.152-.439 1.591 0L21.75 12M4.5 9.75v10.125c0 .621.504 1.125 1.125 1.125H9.75v-4.875c0-.621.504-1.125 1.125-1.125h2.25c.621 0 1.125.504 1.125 1.125V21h4.125c.621 0 1.125-.504 1.125-1.125V9.75M8.25 21h8.25"
            strokeLinecap="round"
            strokeLinejoin="round"
          />
        </svg>
      </div>
      <div
        className="menuLabel MuiBox-root css-0"
      >
        Home
      </div>
    </div>
    <div
      className="Menu MuiBox-root css-0"
    >
      <div
        className="menuIcon MuiBox-root css-0"
      >
        <svg
          className="menuIcon"
          fill="none"
          stroke="currentColor"
          strokeWidth={1.5}
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M15.75 6a3.75 3.75 0 11-7.5 0 3.75 3.75 0 017.5 0zM4.501 20.118a7.5 7.5 0 0114.998 0A17.933 17.933 0 0112 21.75c-2.676 0-5.216-.584-7.499-1.632z"
            strokeLinecap="round"
            strokeLinejoin="round"
          />
        </svg>
      </div>
      <div
        className="menuLabel MuiBox-root css-0"
      >
        Profil
      </div>
    </div>
    <div
      className="Menu MuiBox-root css-0"
    >
      <div
        className="menuIcon MuiBox-root css-0"
      >
        <svg
          className="menuIcon"
          fill="none"
          stroke="currentColor"
          strokeWidth={1.5}
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M12 7.5h1.5m-1.5 3h1.5m-7.5 3h7.5m-7.5 3h7.5m3-9h3.375c.621 0 1.125.504 1.125 1.125V18a2.25 2.25 0 01-2.25 2.25M16.5 7.5V18a2.25 2.25 0 002.25 2.25M16.5 7.5V4.875c0-.621-.504-1.125-1.125-1.125H4.125C3.504 3.75 3 4.254 3 4.875V18a2.25 2.25 0 002.25 2.25h13.5M6 7.5h3v3H6v-3z"
            strokeLinecap="round"
            strokeLinejoin="round"
          />
        </svg>
      </div>
      <div
        className="menuLabel MuiBox-root css-0"
      >
        Articles
      </div>
    </div>
    <div
      className="Menu MuiBox-root css-0"
    >
      <div
        className="menuIcon MuiBox-root css-0"
      >
        <svg
          className="menuIcon"
          fill="none"
          stroke="currentColor"
          strokeWidth={1.5}
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M15.75 9V5.25A2.25 2.25 0 0013.5 3h-6a2.25 2.25 0 00-2.25 2.25v13.5A2.25 2.25 0 007.5 21h6a2.25 2.25 0 002.25-2.25V15M12 9l-3 3m0 0l3 3m-3-3h12.75"
            strokeLinecap="round"
            strokeLinejoin="round"
          />
        </svg>
      </div>
      <div
        className="menuLabel MuiBox-root css-0"
      >
        Logout
      </div>
    </div>
  </div>
</div>
`;
